GithubProfileWiki/Networking/RequestHandler+Extension.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  RequestHandler+Extension.swift
+//  GithubProfileWiki
+//
+//  Created by ilter on 23.01.2022.
+//
+
+import Foundation
+
+extension RequestHandlerProtocol {
+    func setQueryParams(parameters: [String: Any], url: URL) -> URL {
```

Comment:
i am just curious about that, why did you do that with protocol extension instead of static funcs with a util class?

Reply:
Yes you are right, I could like this but, `final class APIService<T: APIHandler>` has  **_RequestHandlerProtocol_** and **_ResponseHandlerProtocol_** and **_setQueryParams & setDefaultHeaderParams_** have the same responsibility for most API requests cases so that I create a default func in it that wraps whole the request from APIService

It feels fancier and more efficient instead of having another class and file to me. Do you have another suggestion or am I wrong? Teach me master yoda :)